engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -842,6 +878,707 @@ static int SetTexture(lua_State* L)
+ *
+ * * `uvs`
+ * : [type:table] a list of the uv coordinates in texture space of the geometry in the form of {u0, v0, u1, v1, ..., un, vn}
+ *
+ * * `indices`
+ * : [type:table] a list of the indices of the geometry in the form {i0, i1, i2, ..., in}. Each tripe in the list represents a triangle.
+ *
+ * @note The index values are zero based where zero refers to the first entry of the vertex and uv lists
+ *
+ * @examples
+ * Add a new animation to an existing atlas
+ *
+ * ```lua
+ * function init(self)
+ *     local data = resource.get_atlas("/main/my_atlas.a.texturesetc")
+ *     local my_animation = {
+ *         id          = "my_new_animation",
+ *         width       = 128,
+ *         height      = 128,
+ *         frame_start = 1,
+ *         frame_end   = 6,
+ *         playback    = go.PLAYBACK_LOOP_PINGPONG,
+ *         fps         = 8
+ *     }
+ *     table.insert(data.animations, my_animation)
+ *     resource.set_atlas("/main/my_atlas.a.texturesetc", data)
+ * end
+ * ```
+ *
+ * @examples
+ * Sets atlas data for a 256x256 texture with a single animation being rendered as a quad
+ *
+ * ```lua
+ * function init(self)
+ *     local params = {
+ *         texture = "/main/my_256x256_texture.texturec",
+ *         animations = {
+ *             {
+ *                 id          = "my_animation",
+ *                 width       = 256,
+ *                 height      = 256,
+ *                 frame_start = 1,
+ *                 frame_end   = 2,
+ *             }
+ *         },
+ *         geometries = {
+ *             {
+ *                 vertices = {
+ *                     0,   0,
+ *                     0,   256,
+ *                     256, 256,
+ *                     256, 0
+ *                 },
+ *                 uvs = {
+ *                     0, 0,
+ *                     0, 256,
+ *                     256, 256,
+ *                     256, 0
+ *                 },
+ *                 indices = { 0,1,2,0,2,3 }
+ *             }
+ *         }
+ *     }
+ *     resource.set_atlas("/main/test.a.texturesetc", params)
+ * end
+ * ```
+ */
+
+static int SetAtlas(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 0);
+
+    dmhash_t path_hash = dmScript::CheckHashOrString(L, 1);
+    void* atlas_res    = CheckResource(L, g_ResourceModule.m_Factory, path_hash, "texturesetc");
+    assert(atlas_res);
+
+    dmGameSystemDDF::TextureSet texture_set_ddf = {};
+    uint32_t frame_index_count                  = 0;
+    uint32_t num_geometries                     = 0;
+    uint32_t num_animations                     = 0;
+    float tex_width                             = 0;
+    float tex_height                            = 0;
+
+    luaL_checktype(L, 2, LUA_TTABLE);
+    lua_pushvalue(L, 2);
+
+    // Set texture resource path
+    lua_getfield(L, -1, "texture");
+    {
+        const char* texture_path         = luaL_checkstring(L, -1);
```

Comment:
I'm wondering if we should support hashes later on. That would mean we would have to expose the "resource.get_path_hash(string)". For now, I'm happy with the way it is. Less code this way.